azure/scope/cluster.go

```diff
@@ -310,7 +310,7 @@ func (s *ClusterScope) RouteTableSpecs() []azure.ResourceSpecGetter {
 			specs = append(specs, &routetables.RouteTableSpec{
 				Name:           subnet.RouteTable.Name,
 				Location:       s.Location(),
-				ResourceGroup:  s.ResourceGroup(),
+				ResourceGroup:  s.Vnet().ResourceGroup,
```

Comment:
The first discrepancy I noticed was that the e2e test sets up the vnet with the route table and NSGs in the vnet resource group, but here we create them in the main resource group. Which one is correct?

cc @CecileRobertMichon

Reply:
*The first and only it seems.

Reply:
Is this potentially a breaking change for existing clusters? what if the route table was previously created in the cluster RG?

Reply:
Vnet resource group gets defaulted to the cluster RG in the webhooks, so I don't think would result in any changes to existing clusters.

Reply:
NSG Spec is also using `s.Vnet().ResourceGroup` so this does look like a bug